### DISCUSSION TITLE
My extensions are not activated by events

### DISCUSSION THREAD

**Discussion by overdev-l** [microsoft/vscode-discussions#460] [Extension Development QnA] [answered]
My extension listens `*` and `onStartupFinished `and `onCommand: deploy.deploy` But when I open my vsocde to select the workspace, I don't receive any prompt from the plugin, he should prompt me. I tried uninstalling the plugin and reinstalling it, but that didn't have any effect
![image](https://user-images.githubusercontent.com/50040262/218663561-0a80e46b-90ba-48dc-81a2-4378ea098c99.png)
![image](https://user-images.githubusercontent.com/50040262/218663604-f3c09aaa-40fa-4fb5-b6d8-7ce07b4b39ae.png)

**Comment by overdev-l** [comment]
![image](https://user-images.githubusercontent.com/50040262/218665124-89b29a28-57ec-4f75-8eeb-33e5783ca711.png)
当我输入命令时提示

**Comment by alefragnani** [comment, marked as answer]
Hi @i-coder-keller ,

Starting in release 1.74, you don't need to declare `onCommand` in activation events anymore ([more details](https://code.visualstudio.com/updates/v1_74#_implicit-activation-events-for-declared-extension-contributions)), so you could replace your `activationEvents` to:

```js
    "activationEvents": [
        "onStartupFinished"
    ],
```

If your commands are properly _registered_ (using `registerCommand`), this error is probably happening _before_, right after the `activate` is called. Something like:


```ts
export async function activate(context: vscode.ExtensionContext) {

    Container.initialize(context);

    // ... some code related to the extension itself, not each command
    // but for some reason, this code is failing, and the `registerCommands` won't be executed
    PathUtils.setExtensionContext(context);
    locators = new Locators();
    const projectStorage: ProjectStorage = new ProjectStorage(getProjectFilePath());

    // register the commands itself
    context.subscriptions.push(vscode.commands.registerCommand("_projectManager.openFolderWelcome", () => {
        const openFolderCommand = isWindows || isMacOS ? "workbench.action.files.openFolder" : "workbench.action.files.openFileFolder"
        vscode.commands.executeCommand(openFolderCommand)
    }));
```

I would suggest you to debug the `activate` event itself and/or look at `Console` window in `Developer: Toggle Developer Tools` command, which could tell you what is failing _before_ the commands are registered. Maybe you have codes like `vscode.window.activeTextEditor.document` in It while debugging, but when you installed the `vsix` file and started the new window, you are on an empty workspace.

Hope this helps

**Reply by overdev-l** [reply]
Thank you very much, I found that there is a production dependency put into the development dependency, thank you very much for your answer
